src/windows/editor_bottom_toolbar.c

```diff
@@ -244,7 +244,7 @@ static void sub_66F6E3()
 		gResearchedRideTypes[i] = 0xFFFFFFFF;
 	}
 
-	for (int i = 0; i < 128; i++) {
+	for (int i = 0; i < countof(RideTypePossibleTrackConfigurations); i++) {
```

Comment:
oh, `gResearchedTrackTypesA` is 128-long. I can add another loop right after this one to initialise them to -1